File: rb/features.rb
# StitchAiDesign SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module StitchAiDesignFeatures
  def self.make_feature(name)
    case name
    when "base"
      StitchAiDesignBaseFeature.new
    when "test"
      StitchAiDesignTestFeature.new
    else
      StitchAiDesignBaseFeature.new
    end
  end
end

File: py/stitchaidesign_sdk/features.py
# StitchAiDesign SDK feature factory

from stitchaidesign_sdk.feature.base_feature import StitchAiDesignBaseFeature
from stitchaidesign_sdk.feature.test_feature import StitchAiDesignTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StitchAiDesignBaseFeature(),
        "test": lambda: StitchAiDesignTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
